RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons/fr.free.nrw.commons.upload.ShareActivity}: java.lang.IllegalArgumentException: invalid provider: null

Posted by 关彳山 on 12 Jul:

```
USER_COMMENT=choose pictures to upload 
ANDROID_VERSION=5.1.1
APP_VERSION_NAME=1.17
BRAND=samsung
PHONE_MODEL=SM-A7100
CUSTOM_DATA=
STACK_TRACE=java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons/fr.free.nrw.commons.upload.ShareActivity}: java.lang.IllegalArgumentException: invalid provider: null
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3134)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3244)
at android.app.ActivityThread.access$1000(ActivityThread.java:219)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1719)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:145)
at android.app.ActivityThread.main(ActivityThread.java:6922)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
Caused by: java.lang.IllegalArgumentException: invalid provider: null
at android.location.LocationManager.checkProvider(LocationManager.java:1742)
at android.location.LocationManager.requestLocationUpdates(LocationManager.java:459)
at fr.free.nrw.commons.upload.GPSExtractor.registerLocationManager(GPSExtractor.java:60)
at fr.free.nrw.commons.upload.GPSExtractor.getCoords(GPSExtractor.java:95)
at fr.free.nrw.commons.upload.ShareActivity.getLocationData(ShareActivity.java:333)
at fr.free.nrw.commons.upload.ShareActivity.onCreate(ShareActivity.java:277)
at android.app.Activity.performCreate(Activity.java:6609)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1134)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3087)
... 10 more
java.lang.IllegalArgumentException: invalid provider: null
at android.location.LocationManager.checkProvider(LocationManager.java:1742)
at android.location.LocationManager.requestLocationUpdates(LocationManager.java:459)
at fr.free.nrw.commons.upload.GPSExtractor.registerLocationManager(GPSExtractor.java:60)
at fr.free.nrw.commons.upload.GPSExtractor.getCoords(GPSExtractor.java:95)
at fr.free.nrw.commons.upload.ShareActivity.getLocationData(ShareActivity.java:333)
at fr.free.nrw.commons.upload.ShareActivity.onCreate(ShareActivity.java:277)
at android.app.Activity.performCreate(Activity.java:6609)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1134)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3087)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3244)
at android.app.ActivityThread.access$1000(ActivityThread.java:219)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1719)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:145)
at android.app.ActivityThread.main(ActivityThread.java:6922)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
```

--- a/app/src/main/java/fr/free/nrw/commons/upload/GPSExtractor.java
+++ b/app/src/main/java/fr/free/nrw/commons/upload/GPSExtractor.java
@@ -57,10 +57,14 @@ public class GPSExtractor {
         provider = locationManager.getBestProvider(criteria, true);
         myLocationListener = new MyLocationListener();
 
-        locationManager.requestLocationUpdates(provider, 400, 1, myLocationListener);
-        Location location = locationManager.getLastKnownLocation(provider);
-        if (location != null) {
-            myLocationListener.onLocationChanged(location);
+        try {
+            locationManager.requestLocationUpdates(provider, 400, 1, myLocationListener);
+            Location location = locationManager.getLastKnownLocation(provider);
+            if (location != null) {
+                myLocationListener.onLocationChanged(location);
+            }
+        } catch (IllegalArgumentException e){
+            Log.e(TAG, "Illegal argument exception", e);
         }
     }
 

